pyuvm/s18_register_model.py

```diff
@@ -0,0 +1,85 @@
+from pyuvm import uvm_object
+
+
+# 18.1.1 Class declaration
+class uvm_reg_block(uvm_object):
+    pass
+
+
+# 18.2.1 Class declaration
+class uvm_reg_map(uvm_object):
+    pass
+
+
+# 18.4.1 Class declaration
+class uvm_reg(uvm_object):
+
+    # 18.3.2.1
+    # TODO Fix signature
+    def __init__(self, name=""):
+        super().__init__(name)
+        self._parent = None
+
+    # 18.4.2.2
+    def configure(self, parent):
+        self._parent = parent
+
+    # 18.4.3.1
+    def get_parent(self):
+        return self._parent
+
+
+# 18.5.1 Class declaration
+class uvm_reg_field(uvm_object):
+
+    # 18.5.3.1
+    def __init__(self, name='uvm_reg_field'):
+        super().__init__(name)
+        self._parent = None
+        self._size = None
+        self._lsb_pos = None
+        self._access = None
+        self._is_volatile = None
+        self._reset = None
+
+    # 18.5.3.2
+    # TODO Fix signature
+    def configure(self, parent, size, lsb_pos, access, is_volatile, reset):
+        # TODO Add validation of arguments
+        # TODO Support binary and hex values for 'reset'
+        self._parent = parent
+        self._size = size
+        self._lsb_pos = lsb_pos
+        self._access = access
+        self._is_volatile = is_volatile
+        self._reset = reset
+
+    # 18.5.4.1
+    def get_parent(self):
```

Comment:
Maybe one can add:
```python3
parent = property(get_parent)
````
so it is more Python-like access avalaible?

Reply:
I made a design decision keep the `get_*` rather than convert them to properties. 

I've never see the `property()` function, I've always used the decorator. 

What does the function do?

Reply:
You can keep `get_*` I guess make sense for compatibility.

If you add `property(get_parent)` one can do `obj.parent` also and this will call `obj.get_parent()`. 

It is up to you. Bit matter of teste.

Some more: https://www.programiz.com/python-programming/property

Reply:
Thanks for the link.  That's excellent in that it keeps the get_* and set_* that the UVM programmer expects but also supports the Python writer's instinct.  

If we add it we should do it across the board in one release so that the user isn't frustrated by finding that some set/get pairs don't have a property.